### ISSUE TITLE
CURRENT_TIMESTAMP not quoted even for non-datetime columns

### ISSUE THREAD

**Issue by MasterOdin** [cakephp/phinx#1891]
The following code will throw an error:

```php
        $this->table('test')
            ->addColumn('test_col', 'string', ['default' => 'CURRENT_TIMESTAMP'])
            ->create();
```

This is because the default value will not be quoted. We can probably update the if check for `CURRENT_TIMESTAMP` either apply if the column type is a datetime/timestamp, or not if the column is a string. I would probably lean towards doing the former, and then for custom types, the user will just be expected to use a `Literal`.

**Comment by dereuromark** [comment]
is that solved through https://github.com/cakephp/phinx/pull/1882 ?

**Comment by MasterOdin** [comment]
No. It'll actually make the problem "worse" in that now anything starting with `CURRENT_TIMESTAMP` would fail (e.g. `CURRENT_TIMESTAMP IS A DUMB DEFAULT`). Fine to complete and merge #1882 and solve this afterwards I think.

**Comment by dereuromark** [comment]
@MasterOdin 
What you think about https://github.com/cakephp/migrations/pull/923 ?
